lib/internal/bootstrap/web/exposed-window-or-worker.js

```diff
@@ -57,32 +57,21 @@ defineReplaceableLazyAttribute(globalThis, 'perf_hooks', ['performance']);
 const { installObjectURLMethods } = require('internal/url');
 installObjectURLMethods();
 
-{
-  // https://fetch.spec.whatwg.org/#fetch-method
-  function set(value) {
-    ObjectDefineProperty(globalThis, 'fetch', {
-      __proto__: null,
-      writable: true,
-      value,
-    });
-  }
-  ObjectDefineProperty(globalThis, 'fetch', {
-    __proto__: null,
-    configurable: true,
-    enumerable: true,
-    set,
-    get() {
-      function fetch(input, init = undefined) {
-        // Loading undici alone lead to promises which breaks lots of tests so we
-        // have to load it really lazily for now.
-        const { fetch: impl } = require('internal/deps/undici/undici');
-        return impl(input, init);
-      }
-      set(fetch);
-      return fetch;
-    },
-  });
-}
+let fetchImpl;
+// https://fetch.spec.whatwg.org/#fetch-method
+ObjectDefineProperty(globalThis, 'fetch', {
+  __proto__: null,
+  configurable: true,
+  enumerable: true,
+  writable: true,
+  value: function value(input, init = undefined) {
```

Comment:
isn't `init = undefined` redundant here?

Reply:
I think it's to specify that the `init` argument is optional.

Reply:
right, but you're defaulting it to undefined -.which would be the case if it wasn't defined, it's redundant

Reply:
Same comment for https://github.com/nodejs/node/blob/main/deps/undici/undici.js#L12497

It makes no sense to default an argument value to undefined ever

Reply:
It affects the `.length` of the function, which is observable. Parameters with default values don't count towards the length. And since the correct length is 1, not 2, this is the only way to do that without either explicitly setting the length attribute or referencing `arguments`.